tests/test_dask_image/test_ndmeasure/test_core.py

```diff
@@ -366,6 +366,97 @@ def test_label(seed, prob, shape, chunks, connectivity):
     _assert_equivalent_labeling(a_l, d_l.compute())
 
 
+a = np.array(
+    [
+        [0, 0, 1, 0, 0, 1, 1, 0, 0, 0],
+        [0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+        [1, 1, 0, 0, 0, 0, 1, 1, 1, 1],
+        [1, 1, 0, 0, 0, 0, 1, 1, 1, 1],
+        [1, 0, 0, 0, 1, 0, 1, 1, 1, 0],
+        [0, 1, 0, 0, 1, 0, 1, 1, 1, 0],
+        [0, 0, 1, 0, 1, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        [0, 0, 1, 0, 0, 1, 1, 0, 0, 0],
+    ]
+)
+
+
+@pytest.mark.parametrize(
+    "a, a_res, wrap_axes",
+    [
+        (
+            a,
+            np.array(
+                [
+                    [0, 0, 1, 0, 0, 3, 3, 0, 0, 0],
+                    [0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+                    [1, 1, 0, 0, 0, 0, 1, 1, 1, 1],
+                    [1, 1, 0, 0, 0, 0, 1, 1, 1, 1],
+                    [1, 0, 0, 0, 2, 0, 1, 1, 1, 0],
+                    [0, 1, 0, 0, 2, 0, 1, 1, 1, 0],
+                    [0, 0, 1, 0, 2, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 4, 0, 0, 5, 5, 0, 0, 0],
+                ]
+            ),
+            (0,),
+        ),
+        (
+            a,
+            np.array(
+                [
+                    [0, 0, 1, 0, 0, 3, 3, 0, 0, 0],
+                    [0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+                    [1, 1, 0, 0, 0, 0, 4, 4, 4, 4],
+                    [1, 1, 0, 0, 0, 0, 4, 4, 4, 4],
+                    [1, 0, 0, 0, 2, 0, 4, 4, 4, 0],
+                    [0, 1, 0, 0, 2, 0, 4, 4, 4, 0],
+                    [0, 0, 1, 0, 2, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 0, 0, 3, 3, 0, 0, 0],
+                ]
+            ),
+            (1,),
+        ),
+        (
+            a,
+            np.array(
+                [
+                    [0, 0, 1, 0, 0, 3, 3, 0, 0, 0],
+                    [0, 0, 1, 0, 0, 0, 0, 0, 0, 0],
+                    [1, 1, 0, 0, 0, 0, 1, 1, 1, 1],
+                    [1, 1, 0, 0, 0, 0, 1, 1, 1, 1],
+                    [1, 0, 0, 0, 2, 0, 1, 1, 1, 0],
+                    [0, 1, 0, 0, 2, 0, 1, 1, 1, 0],
+                    [0, 0, 1, 0, 2, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 0, 1, 0, 0, 3, 3, 0, 0, 0],
+                ]
+            ),
+            (0, 1),
+        ),
+        pytest.param(
+            np.array([[1, 0, 0], [0, 0, 0], [0, 0, 1]]),
+            np.array([[1, 0, 0], [0, 0, 0], [0, 0, 1]]),
+            (0, 1),
+            marks=pytest.mark.xfail(reason="Can't wrap corner labels"),
+        ),
+    ],
+)
+def test_label_wrap(a, a_res, wrap_axes):
+    d = da.from_array(a, chunks=(5, 5))
+
+    s = np.ones((3, 3))
```

Comment:
As jni [commented](https://github.com/dask/dask-image/pull/344#issuecomment-1758799104), using structuring elements with connectivity > 1 would lead to problems in the corners. `scipy.ndimage.morphology.generate_binary_structure` is a nice convenience function for creating structuring elements.